osu.Game/Screens/Play/BreakOverlay.cs

```diff
@@ -96,60 +104,78 @@ public BreakOverlay(bool letterboxing)
                         Anchor = Anchor.Centre,
                         Origin = Anchor.Centre,
                     }
-                }
-            };
-        }
-
-        protected override void LoadComplete()
-        {
-            base.LoadComplete();
-            initializeBreaks();
-        }
-
-        private void initializeBreaks()
-        {
-            if (!IsLoaded) return; // we need a clock.
-
-            FinishTransforms(true);
-            Scheduler.CancelDelayedTasks();
-
-            if (breaks == null) return; //we need breaks.
-
-            foreach (var b in breaks)
-            {
-                if (!b.HasEffect)
-                    continue;
+          }
+      };
+    }
 
-                using (BeginAbsoluteSequence(b.StartTime, true))
-                {
-                    fadeContainer.FadeIn(fade_duration);
-                    breakArrows.Show(fade_duration);
+    protected override void LoadComplete()
+    {
+      base.LoadComplete();
+      initializeBreaks();
+    }
+    [BackgroundDependencyLoader]
+    private void load(OsuConfigManager config)
+    {
+      // We need BackgroundOpacityOffset to lighten during breaks.
+      if (BackgroundOpacityOffset == null) return;
```

Comment:
I'd suggest to either passing the `BackgroundOpacityOffset` in ctor or assume a default so it still works if the Bindable is set later.

Reply:
I could remove that line and add a check like this:
```c#
private void updateBackgroundOpacityOffset()
{
    if (BackgroundOpacityOffset != null)
    {
      this.TransformBindableTo(
          BackgroundOpacityOffset,
          lightenDuringBreaks.Value && currentlyOnBreak.Value ? 1 : 0,
          fade_duration,
          Easing.OutQuint
      );
    }
}
```